data/data/bootstrap/files/usr/local/bin/kubelet.sh.template

```diff
@@ -11,6 +11,10 @@
     --runtime-request-timeout="${KUBELET_RUNTIME_REQUEST_TIMEOUT}" \
 {{- if .UseIPv6ForNodeIP }}
     --node-ip=:: \
+{{- end}}
+{{- if .BootstrapNodeIp }}
+    --node-ip="{{.BootstrapNodeIp}}" \
+    --address="{{.BootstrapNodeIp}}" \
```

Comment:
Why not just set `--node-ip=0.0.0.0` and listen on all (as we currently do for IPv6)?

Also why do you set  --address="{{.BootstrapNodeIp}} ?

Reply:
If you set 0.0.0.0 it is the same as not setting anything.
The idea of setting "::" in ipv6 is to say to kubelet to choose ip from ipv6 and not to use ipv4

Reply:
kube-api always listen on 0.0.0.0 it just advertise itself on different one 
example :

[root@test-infra-cluster-86ad0a59-master-2 core]# oc get endpoints 
NAME         ENDPOINTS             AGE
kubernetes   192.168.127.12:6443   4m32s
[root@test-infra-cluster-86ad0a59-master-2 core]# netstat -tulpn | grep 6443
tcp6       0      0 :::6443                 :::*                    LISTEN      12231/kube-apiserve

Reply:
I'm not sure what the above means apart from seeing that kube-apiserver is listening on all IPv6 (as you would expect when it's configured with `::`)

Reply:
kube-api always listen on all interfaces, the issue is which one it set in entrypoint. 
Regarding the example:
I set machine cidr to 192.168.145.0/24 and set node-ip on kubelet service to 0.0.0.0, what we got is kubeapi endpoint set to 192.168.127.10 and kube-api (as always) listen on all

Reply:
So if you set node-ip (in kubelet) to the IP on that machine cidr that will be the IP for the kubeapi endpoint?

Reply:
exactly